### ISSUE TITLE
BUG: scipy.optimize.linprog exits the entire program without raising an assertion that can be catched

### ISSUE THREAD

**Issue by iincer** [scipy/scipy#17521]
### Describe your issue.

There are inputs (like the one shown in the example below) that make scipy.optimize.linprog exit on a failed c++ assertion.
This exception cannot be catched in python. The program just exits.

### Reproducing Code Example

```python
from scipy.optimize import linprog
import numpy as np

objective = np.array([1, 1,  0, -1,  0,  0])
a_opt = np.array(
[[-1, -1,  1,  0,  0,  0],
 [-1, -1,  0,  1,  0,  0],
 [ 0,  0,  0, -1,  4,  1],
 [ 0,  1,  0, -1,  4,  0],
 [ 1,  0,  0,  0,  0,  0],
 [ 0,  0,  0,  0,  5,  0]])
b_opt = np.array(
[0, 1, 0, 0, 2, 1])

res = linprog(c=objective, A_ub=a_opt, b_ub=b_opt, bounds=(None, None))
```

### Error message

```shell
Assertion failed: model_status_ == HighsModelStatus::kInfeasible || model_status_ == HighsModelStatus::kUnbounded, file ../../scipy/_lib/highs/src/lp_data/Highs.cpp, line 1125
```

### SciPy/NumPy/Python version information

1.9.3 1.23.5 sys.version_info(major=3, minor=10, micro=0, releaselevel='final', serial=0)

**Comment by WarrenWeckesser** [comment]
Thanks for reporting the issue, @iincer.  I have confirmed the crash, and also confirmed that the problem has been fixed by https://github.com/scipy/scipy/pull/17430, so I'm closing the issue.  The fix will be in SciPy 1.10.0.

**Comment by iincer** [comment]
Perfect. Thank you, @WarrenWeckesser.
